engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/thread/SleepNode.java

```diff
@@ -0,0 +1,52 @@
+package org.enso.interpreter.node.expression.builtin.thread;
+
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.control.ThreadInterruptedException;
+import org.enso.interpreter.runtime.error.PanicException;
+
+@BuiltinMethod(
+    type = "Thread",
+    name = "sleep",
+    description = "Sleep in the current thread.",
+    autoRegister = false,
+    inlineable = true)
+public abstract class SleepNode extends Node {
+
+  public abstract Object execute(Object timeInMilliseconds);
+
+  public static SleepNode build() {
+    return SleepNodeGen.create();
+  }
+
+  @Specialization
+  Object doLong(long timeInMilliseconds) {
+    try {
+      Thread.sleep(timeInMilliseconds);
+      return EnsoContext.get(this).getBuiltins().nothing();
+    } catch (InterruptedException e) {
+      throw new ThreadInterruptedException();
+    }
+  }
+
+  @Specialization
+  Object doDouble(double timeInMilliseconds) {
+    try {
+      Thread.sleep(Double.valueOf(timeInMilliseconds).longValue());
+      return EnsoContext.get(this).getBuiltins().nothing();
+    } catch (InterruptedException e) {
+      throw new ThreadInterruptedException();
+    }
+  }
```

Comment:
Should we raise an error if the fractional part is non-zero?
Allowing `Thread.sleep 0.5` may suggest like it's possible to sleep for a fraction of a millisecond, but this will actually convert into `Thread.sleep 0`.

Reply:
I don't think anyone is going to measure that plus it is a private method.